### PR TITLE
Allow to configure prometheus retention time and size

### PR DESCRIPTION
**Pull request by linki**
It's handy to tweak this setting to keep Prometheus disk usage under control.

### CODE REVIEW

**Comment by linki** [comment]
Or we just go with the retention-size flag.

**Comment by szuecs** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
IMO if the goal is to control the data size we should steer it explicitly. The docs https://prometheus.io/docs/prometheus/latest/storage/#operational-aspects do not explicitly prohibit having two options at the same time and docker seems to start just fine with both:
```
docker run --rm -it prom/prometheus --storage.tsdb.retention.time=1d --storage.tsdb.retention.size=49GB --config.file=/etc/prometheus/prometheus.yml
```

From the code it looks like they remove blocks that are both beyond retention time and size (i.e. use OR logic) https://github.com/prometheus/prometheus/blob/b63d7c786e045385b1abdff6fe7e5ac3b6f13a7a/tsdb/db.go#L1191-L1197

File: test/e2e/cluster_config.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ clusters:
     nlb_switch: "pre"
     vm_dirty_bytes: 134217728
     vm_dirty_background_bytes: 67108864
-    prometheus_tsdb_retention_size: enabled
+    prometheus_retention_size: 1258292KB
```
Comment (linki) [inline test/e2e/cluster_config.sh]: This is the value that would have been used for test clusters, the only user of this config item.

Comment (linki) [reply test/e2e/cluster_config.sh]: Changed to the more readable 1280MB.

File: test/e2e/cluster_config.sh [hunk 2]
```diff
@@ -35,7 +35,7 @@ clusters:
     nlb_switch: "pre"
     vm_dirty_bytes: 134217728
     vm_dirty_background_bytes: 67108864
-    prometheus_tsdb_retention_size: enabled
+    prometheus_retention_size: 1280MB
```
Comment (linki) [inline test/e2e/cluster_config.sh]: This is the value that would have been used for test clusters, the only user of this config item (to be exact: 1258292KB)

Comment (AlexanderYastrebov) [reply test/e2e/cluster_config.sh]: I think this is used for e2e test clusters, the actual value probably does not make any difference as these clusters are short-lived so maybe it should be just dropped and rely on defaults

**Comment by mikkeloscar** [comment]
:+1:

**Comment by linki** [comment]
We decided to drop the custom TSDB size limit for e2e clusters because we don't fully understand why it's needed.

@mikkeloscar If you happen to remember, feel free to comment (see related PR https://github.com/zalando-incubator/kubernetes-on-aws/pull/2999). Otherwise we'll go ahead with this approach.

**Comment by mikkeloscar** [comment]
@linki @AlexanderYastrebov I agree and also don't see a need for this special config in e2e.

**Comment by mikkeloscar** [comment]
:+1:

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**